File: backend/models/announcement.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Announcement struct {
	ID              uint           `gorm:"primaryKey" json:"id"`
	OwnerID         uint           `json:"owner_id"`
	Owner           User           `gorm:"foreignKey:OwnerID" json:"owner,omitempty"`
	BoardingHouseID uint           `json:"boarding_house_id"`
	BoardingHouse   BoardingHouse  `gorm:"foreignKey:BoardingHouseID" json:"boarding_house,omitempty"`
	TargetType      string         `gorm:"size:20;default:boarding_house" json:"target_type"` // boarding_house, user
	TargetUserID    *uint          `json:"target_user_id"`
	TargetUser      *User          `gorm:"foreignKey:TargetUserID" json:"target_user,omitempty"`
	Title           string         `gorm:"size:100" json:"title"`
	Content         string         `gorm:"type:text" json:"content"`
	Date            time.Time      `json:"date"`
	Icon            string         `gorm:"size:50" json:"icon"` // water, electric, repair, info
	CreatedAt       time.Time      `json:"created_at"`
	UpdatedAt       time.Time      `json:"updated_at"`
	DeletedAt       gorm.DeletedAt `gorm:"index" json:"-"`
}

File: rumatenant/lib/models/announcement.dart
class Announcement {
  final int id;
  final int ownerId;
  final int boardingHouseId;
  final String targetType;
  final int? targetUserId;
  final String title;
  final String content;
  final DateTime date;
  final String icon;
  final DateTime createdAt;

  Announcement({
    required this.id,
    required this.ownerId,
    required this.boardingHouseId,
    this.targetType = 'boarding_house',
    this.targetUserId,
    required this.title,
    required this.content,
    required this.date,
    this.icon = 'info',
    required this.createdAt,
  });

  factory Announcement.fromJson(Map<String, dynamic> json) {
    return Announcement(
      id: json['id'] ?? 0,
      ownerId: json['owner_id'] ?? 0,
      boardingHouseId: json['boarding_house_id'] ?? 0,
      targetType: json['target_type'] ?? 'boarding_house',
      targetUserId: json['target_user_id'],
      title: json['title'] ?? '',
      content: json['content'] ?? '',
      date: json['date'] != null
          ? DateTime.parse(json['date'])
          : DateTime.now(),
      icon: json['icon'] ?? 'info',
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'])
          : DateTime.now(),
    );
  }
}
